### PR TITLE
DOC: `MultiIndex.get_level_values` mentions implicit type casting if level contains null.

### LINKED ISSUE #44792: BUG: MultiIndex constructed with nulls inconsistent with `dtypes` and `get_level_values`
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
In [40]: pd.MultiIndex.from_arrays([[1, None, 2], [None, None, 3]]).dtypes
Out[40]: 
level_0    int64
level_1    int64
dtype: object
In [43]: pd.MultiIndex.from_arrays([[1, None, 2], [None, None, 3]]).get_level_values(0)
Out[43]: Float64Index([1.0, nan, 2.0], dtype='float64')
In [44]: pd.MultiIndex.from_arrays([[1, None, 2], [None, None, 3]]).get_level_values(1)
Out[44]: Float64Index([nan, nan, 3.0], dtype='float64')
```

### Issue Description

`dtypes` reports that both levels are of integer types, but the returned level values are float types.

### Expected Behavior

Either`dtypes` reports float types or `dtypes` reports nullable integer types and `get_level_values` returns that.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5
python           : 3.8.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.11.0-40-generic
Version          : #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8

pandas           : 1.3.4
numpy            : 1.20.3
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 59.4.0
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.29.0
sphinx           : 4.3.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.4
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.30.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
fsspec           : 2021.11.1
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.54.1

</details>

### PR DESCRIPTION
**Pull request by isVoid**
- [x] closes #44792 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -1647,7 +1647,9 @@ def get_level_values(self, level):
         """
         Return vector of label values for requested level.
 
-        Length of returned vector is equal to the length of the index.
+        Length of returned vector is equal to the length of the index. Note
+        that if the level contains null, the result may be casted to ``float``
```
Comment (phofl) [inline pandas/core/indexes/multi.py:1651]: missing values instead of null

**Review by phofl — CHANGES_REQUESTED** [review]
Could you mention that this happens because we convert the level to a regular Index instead of a MultiIndex

**Comment by isVoid** [comment]
I was hesitant to write "internally MultiIndex is implemented by codes but regular index doesn't", concerning that reveals too much implementation details in the docstring. Let me know if it fits in.

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -1647,7 +1647,11 @@ def get_level_values(self, level):
         """
         Return vector of label values for requested level.
 
-        Length of returned vector is equal to the length of the index.
+        Length of returned vector is equal to the length of the index. Note
+        that if the level contains missing values, the result may be casted to
+        ``float`` with missing values specified with ``NaN``. This is because
```
Comment (phofl) [inline pandas/core/indexes/multi.py:1652]: as ``NaN``

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -1647,7 +1647,11 @@ def get_level_values(self, level):
...
+        the level is converted to a regular ``Index``, not to a ``MultiIndex``
```
Comment (phofl) [inline pandas/core/indexes/multi.py:1653]: Can stop after Index

**Comment by jreback** [comment]
can you add an example which illustrates this as well.

**Comment by isVoid** [comment]
Example added. @jreback

File: pandas/core/indexes/multi.py [hunk 4]
```diff
@@ -1647,7 +1647,10 @@ def get_level_values(self, level):
         """
         Return vector of label values for requested level.
 
-        Length of returned vector is equal to the length of the index.
+        Length of returned vector is equal to the length of the index. Note
```
Comment (jreback) [inline pandas/core/indexes/multi.py:1650]: can you make this a Note itself, e.g. the added text.  (e.g. sub-section).

**Review by jreback — CHANGES_REQUESTED** [review]
one comment. ping when ready

File: pandas/core/indexes/multi.py (hunk 4 above)
Comment (isVoid) [reply pandas/core/indexes/multi.py]: Done! @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @isVoid

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**